### PR TITLE
Add UserSkillsConsumer

### PR DESCRIPTION
**Pull request by SolovyovDmitry**
Add UserSkillsConsumer to obtain skills data for user data

### CODE REVIEW

File: SkillService.Broker/Consumers/GetUserSkillsConsumer.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    public async Task Consume(ConsumeContext<IGetUserSkillsRequest> context)
+    {
+      List<UserSkillData> userSkills = await GetUserSkillsAsync(context.Message);
+
+      await context.RespondAsync<IOperationResult<IGetUserSkillsResponse>>(
+        OperationResultWrapper.CreateResponse(_ => IGetUserSkillsResponse.CreateObj(userSkills), context));
+    }
+
+    private async Task<List<UserSkillData>> GetUserSkillsAsync(IGetUserSkillsRequest request)
```
Comment (MariyaSin) [inline SkillService.Broker/Consumers/GetUserSkillsConsumer.cs]: plese, move private method before ctor

File: src/SkillService/Startup.cs [hunk 2]
```diff
@@ -182,14 +184,34 @@ private void ConfigureMassTransit(IServiceCollection services)
...
       services.AddMassTransitHostedService();
     }
 
+    private void ConfigureEndpoints(
+      IBusRegistrationContext context, 
+      IRabbitMqBusFactoryConfigurator cfg, 
+      RabbitMqConfig rabbitMqConfig)
+    {
+      //TODO: Check proper endpoint
+      cfg.ReceiveEndpoint(rabbitMqConfig.CheckUserRightsEndpoint, ep =>
```
Comment (MariyaSin) [inline src/SkillService/Startup.cs]: wrong endpoint
add endpoint to appsettings.json and call it throw RabbitMqConfig

File: SkillService.Broker/Consumers/GetUserSkillsConsumer.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+{
+  public class GetUserSkillsConsumer : IConsumer<IGetUserSkillsRequest>
+  {
+    private readonly IUserSkillRepository _repository;
+
+    private async Task<List<UserSkillData>> GetUserSkillsAsync(IGetUserSkillsRequest request)
+    {
+      List<DbUserSkill> userSkills = await _repository.GetUserSkillsAsync(request.UserId);
+
+      return userSkills.Select(
```
Comment (MariyaSin) [inline SkillService.Broker/Consumers/GetUserSkillsConsumer.cs]: add check if userSkills is null (if something went wrong)
return userSkills?.Select(...

File: src/SkillService.Data/Interfaces/IUserSkillRepository.cs [hunk 4]
```diff
@@ -11,6 +11,8 @@ public interface IUserSkillRepository
   {
     Task<List<Guid>> GetAsync(Guid userId);
 
+    Task<List<DbUserSkill>> GetUserSkillsAsync(Guid userId);
```
Comment (MariyaSin) [inline src/SkillService.Data/Interfaces/IUserSkillRepository.cs]: please, rename your method to GetAsync, and set GetUserSkillIdsAsync name to method on 12 line

File: src/SkillService/Startup.cs [hunk 5]
```diff
@@ -182,14 +184,34 @@ private void ConfigureMassTransit(IServiceCollection services)
...
 
       services.AddMassTransitHostedService();
     }
 
+    private void ConfigureEndpoints(
+      IBusRegistrationContext context, 
+      IRabbitMqBusFactoryConfigurator cfg, 
+      RabbitMqConfig rabbitMqConfig)
+    {
+      //TODO: Check proper endpoint
```
Comment (MariyaSin) [inline src/SkillService/Startup.cs]: remove comment

File: SkillService.Broker/SkillService.Broker.csproj [hunk 6]
```diff
@@ -0,0 +1,36 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (MariyaSin) [inline SkillService.Broker/SkillService.Broker.csproj]: inclue target file from "solution items"

**Review by MariyaSin — APPROVED** [review]

File: src/SkillService.Broker/Consumers/GetUserSkillsConsumer.cs [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+using LT.DigitalOffice.SkillService.Models.Db;
+using MassTransit;
+
+namespace LT.DigitalOffice.SkillService.Broker.Consumers
+{
+  public class GetUserSkillsConsumer : IConsumer<IGetUserSkillsRequest>
+  {
+    private readonly IUserSkillRepository _repository;
+
+    private async Task<List<UserSkillData>?> GetUserSkillsAsync(IGetUserSkillsRequest request)
```
Comment (VLiamin) [inline src/SkillService.Broker/Consumers/GetUserSkillsConsumer.cs]: Why '?'. You can write only List'<'...'>', because it is reference type.

**Review by VLiamin — APPROVED** [review]

File: src/SkillService.Broker/Consumers/GetUserSkillsConsumer.cs (hunk 7 above)
Comment (SolovyovDmitry) [reply src/SkillService.Broker/Consumers/GetUserSkillsConsumer.cs]: Seemed strange for me too. Resharper suggestion

**Pull request merged**